Mathlib/Analysis/NormedSpace/Star/ContinuousFunctionalCalculus/Order.lean

```diff
@@ -44,10 +110,86 @@ lemma IsSelfAdjoint.neg_algebraMap_norm_le_self {a : A} (ha : IsSelfAdjoint a :=
     exact IsSelfAdjoint.le_algebraMap_norm_self (neg ha)
   exact neg_le.mp this
 
-lemma mul_star_le_algebraMap_norm_sq {a : A} : a * star a ≤ algebraMap ℝ A (‖a‖ ^ 2) := by
+lemma CstarRing.mul_star_le_algebraMap_norm_sq {a : A} : a * star a ≤ algebraMap ℝ A (‖a‖ ^ 2) := by
   have : a * star a ≤ algebraMap ℝ A ‖a * star a‖ := IsSelfAdjoint.le_algebraMap_norm_self
   rwa [CstarRing.norm_self_mul_star, ← pow_two] at this
 
-lemma star_mul_le_algebraMap_norm_sq {a : A} : star a * a ≤ algebraMap ℝ A (‖a‖ ^ 2) := by
+lemma CstarRing.star_mul_le_algebraMap_norm_sq {a : A} : star a * a ≤ algebraMap ℝ A (‖a‖ ^ 2) := by
   have : star a * a ≤ algebraMap ℝ A ‖star a * a‖ := IsSelfAdjoint.le_algebraMap_norm_self
   rwa [CstarRing.norm_star_mul_self, ← pow_two] at this
+
+lemma IsSelfAdjoint.toReal_spectralRadius_eq_norm {a : A} (ha : IsSelfAdjoint a) :
+    (spectralRadius ℝ a).toReal = ‖a‖ := by
+  simp [ha.spectrumRestricts.spectralRadius_eq, ha.spectralRadius_eq_nnnorm]
+
+lemma CstarRing.norm_or_neg_norm_mem_spectrum [Nontrivial A] {a : A}
+    (ha : IsSelfAdjoint a := by cfc_tac) : ‖a‖ ∈ spectrum ℝ a ∨ -‖a‖ ∈ spectrum ℝ a := by
+  have ha' : SpectrumRestricts a Complex.reCLM := ha.spectrumRestricts
+  have := ha.toReal_spectralRadius_eq_norm
+  convert Real.spectralRadius_mem_spectrum_or (ha'.image ▸ (spectrum.nonempty a).image _)
+    <;> exact id (Eq.symm this)
+
+lemma CstarRing.nnnorm_mem_spectrum_of_nonneg [Nontrivial A] {a : A} (ha : 0 ≤ a := by cfc_tac) :
+    ‖a‖₊ ∈ spectrum ℝ≥0 a := by
+  have : IsSelfAdjoint a := .of_nonneg ha
+  convert NNReal.spectralRadius_mem_spectrum (a := a) ?_ (.nnreal_of_nonneg ha)
+  · simp [this.spectrumRestricts.spectralRadius_eq, this.spectralRadius_eq_nnnorm]
+  · exact this.spectrumRestricts.image ▸ (spectrum.nonempty a).image _
+
+lemma CstarRing.norm_mem_spectrum_of_nonneg [Nontrivial A] {a : A} (ha : 0 ≤ a := by cfc_tac) :
+    ‖a‖ ∈ spectrum ℝ a := by
+  simpa using spectrum.algebraMap_mem ℝ <| CstarRing.nnnorm_mem_spectrum_of_nonneg ha
+
+end Cstar_unital
+
+section Cstar_nonunital
+
+variable {A : Type*} [NonUnitalNormedRing A] [CompleteSpace A] [PartialOrder A] [StarRing A]
+  [StarOrderedRing A] [CstarRing A] [NormedSpace ℂ A] [StarModule ℂ A]
+  [SMulCommClass ℂ A A] [IsScalarTower ℂ A A]
+
+open ComplexOrder in
+instance CstarRing.instNonnegSpectrumClassComplexNonUnital : NonnegSpectrumClass ℂ A where
+  quasispectrum_nonneg_of_nonneg a ha x hx := by
+    rw [Unitization.quasispectrum_eq_spectrum_inr' ℂ ℂ a] at hx
+    exact spectrum_nonneg_of_nonneg (Unitization.inr_nonneg_iff.mpr ha) hx
```

Comment:
Same here

Reply:
This also can't go in `Instances` because of `Unitization.inr_nonneg_iff` (which also must stay in this file according to `#find_home`).

Reply:
fair enough here.